tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -12,45 +22,130 @@ function FileUpload({ section }) {
   // e.g. "1 - Active Case Data" => ["1", "Active Case Data"]
   const [sectionNumber, sectionName] = section.split(' - ')
 
-  const file = files.find((currentFile) => sectionName === currentFile.section)
+  const selectedFile = files.find((file) => sectionName === file.section)
 
-  const formattedSectionName = file.section
+  const formattedSectionName = selectedFile.section
     .split(' ')
     .map((word) => word.toLowerCase())
     .join('-')
 
-  const fileName = file?.fileName
+  const fileName = selectedFile?.fileName
   const hasUploadedFile = Boolean(fileName)
 
   const ariaDescription = hasUploadedFile
-    ? `Selected File ${file?.fileName}. To change the selected file, click this button.`
+    ? `Selected File ${selectedFile?.fileName}. To change the selected file, click this button.`
     : `Drag file here or choose from folder.`
 
-  const uploadFile = ({ target }) => {
-    dispatch(clearError({ section: target.name }))
-    dispatch(
-      upload({
-        file: target.files[0],
-        section: target.name,
-      })
-    )
+  const inputRef = useRef(null)
+
+  const validateAndUploadFile = (event) => {
+    setlocalAlertState({
+      active: false,
+      type: null,
+      message: null,
+    })
+
+    const { name } = event.target
+    const file = event.target.files[0]
+
+    // Clear existing errors and the current
+    // file in the state if the user is re-uploading
+    dispatch(clearError({ section: name }))
+    dispatch(clearFile({ section: name }))
+
+    // Get the the first 4 bytes of the file with which to check file signatures
+    const blob = file.slice(0, 4)
+
+    const input = inputRef.current
+    const dropTarget = inputRef.current.parentNode
+
+    const filereader = new FileReader()
+
+    filereader.onloadend = (evt) => {
+      if (!evt.target.error) {
+        // Read in the file blob "headers: and create a hex string signature
+        const uint = new Uint8Array(evt.target.result)
+        const bytes = []
+        uint.forEach((byte) => {
+          bytes.push(byte.toString(16))
+        })
+        const header = bytes.join('')
+
+        switch (header) {
+          // For some reason, fileType.fromBlob won't detect image/png;
+          // Account for this by checking for png and some other
+          // file signatures manually.
```

Comment:
can we document what file types these are?

Reply:
Done